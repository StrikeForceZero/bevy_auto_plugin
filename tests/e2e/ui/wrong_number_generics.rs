use bevy_auto_plugin::prelude::*;

#[derive(AutoPlugin)]
#[auto_plugin(impl_plugin_trait)]
struct TestPlugin;

#[auto_register_type(plugin = TestPlugin, generics(bool))]
struct Foo<T1, T2>(T1, T2);

// dummy main
fn main() {}
