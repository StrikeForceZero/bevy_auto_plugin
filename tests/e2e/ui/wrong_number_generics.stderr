error[E0107]: struct takes 2 generic arguments but 1 generic argument was supplied
 --> tests/e2e/ui/wrong_number_generics.rs:8:8
  |
7 | #[auto_register_type(plugin = TestPlugin, generics(bool))]
  |                                                    ---- supplied 1 generic argument
8 | struct Foo<T1, T2>(T1, T2);
  |        ^^^ expected 2 generic arguments
  |
note: struct defined here, with 2 generic parameters: `T1`, `T2`
 --> tests/e2e/ui/wrong_number_generics.rs:8:8
  |
8 | struct Foo<T1, T2>(T1, T2);
  |        ^^^ --  --
help: add missing generic argument
  |
7 | #[auto_register_type(plugin = TestPlugin, generics(bool, _))]
  |                                                        +++
